test/lib/samson/secrets/vault_server_test.rb

```diff
@@ -49,6 +49,15 @@
         to_raise: Vault::HTTPError.new("address", stub(code: '200'))
       ) { refute_valid server }
     end
+
+    it "nullifies the association on destroy" do
```

Comment:
kinda vague and focussed on the implementation

`it "removes unfindable servers from associated deploy group"`